### PR TITLE
Add validation for EKS cluster targeted jobs

### PR DESCRIPTION
**Pull request by TerryHowe**
Validate EKS cluster targeted jobs.

### CODE REVIEW

File: config/jobs/kubernetes/release/release-config.yaml [hunk 1]
```diff
@@ -70,6 +70,13 @@ presubmits:
         command:
         - make
         - test
```
Comment (TerryHowe) [inline config/jobs/kubernetes/release/release-config.yaml:72]: I'm not entirely sure what this `make test` does, but seemed like 9G would be plenty

Comment (BenTheElder) [reply config/jobs/kubernetes/release/release-config.yaml:72]: not sure either but we can look at https://prow.k8s.io/?job=pull-release-test => recent green run https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/release/2959/pull-release-test/1635375671163228160 

looks like unit tests, this is a pretty generous allocation for that. we can also merge and iterate

File: config/tests/jobs/jobs_test.go [hunk 2]
```diff
@@ -1115,8 +1121,7 @@ func TestK8sInfraProwBuildJobsCIPolicy(t *testing.T) {
 	jobsToFix := 0
 	jobs := allStaticJobs()
 	for _, job := range jobs {
-		// Only consider Pods destined for the k8s-infra-prow-builds cluster
-		if job.Spec == nil || job.Cluster != "k8s-infra-prow-build" {
+		if job.Spec == nil || (!isCritical(job.Cluster) && !isEKSCluster(job.Cluster)) {
```
Comment (BenTheElder) [inline config/tests/jobs/jobs_test.go:1124]: I think later we should swap this for `isCommunity` or something but right now we seem to be missing the `k8s-infra-prow-builds-trusted` cluster here and that's going to be a big mess, so let's punt.

File: config/tests/jobs/jobs_test.go [hunk 3]
```diff
@@ -52,6 +52,14 @@ var k8sProw = flag.Bool("k8s-prow", true, "If the config is for k8s prow cluster
 // Loaded at TestMain.
 var c *cfg.Config
 
+func isCritical(clusterName string) bool {
```
Comment (BenTheElder) [inline config/tests/jobs/jobs_test.go:55]: unimportant nit: these method names are inconsistent `isEKSCluster` vs `isCritical`.

**Review by BenTheElder — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**